src/libOpenImageIO/imagebufalgo_xform.cpp

```diff
@@ -1144,4 +1155,232 @@ ImageBufAlgo::rotate(const ImageBuf& src, float angle, string_view filtername,
 }
 
 
+
+template<typename DSTTYPE, typename SRCTYPE, typename STTYPE>
+static bool
+st_warp_(ImageBuf& dst, const ImageBuf& src, const ImageBuf& stbuf, int chan_s,
+         int chan_t, bool flip_s, bool flip_t, const Filter2D* filter, ROI roi,
+         int nthreads)
+{
+    OIIO_DASSERT(filter);
+    OIIO_DASSERT(dst.spec().nchannels >= roi.chend);
+
+    ImageBufAlgo::parallel_image(roi, nthreads, [&](ROI roi) {
+        const ImageSpec& srcspec(src.spec());
+        const ImageSpec& dstspec(dst.spec());
+        const int src_width  = srcspec.full_width;
+        const int src_height = srcspec.full_height;
+
+        const float xscale = float(dstspec.full_width) / src_width;
+        const float yscale = float(dstspec.full_height) / src_height;
+
+        const int xbegin = src.xbegin();
+        const int xend   = src.xend();
+        const int ybegin = src.ybegin();
+        const int yend   = src.yend();
+
+        // The horizontal and vertical filter radii, in source pixels.
+        // We will sample and filter the source over
+        //   [x-filterrad_x, x+filterrad_x] X [y-filterrad_y,y+filterrad_y].
+        const int filterrad_x = (int)ceilf(filter->width() / 2.0f / xscale);
+        const int filterrad_y = (int)ceilf(filter->height() / 2.0f / yscale);
+
+        ImageBuf::ConstIterator<SRCTYPE> src_iter(src);
```

Comment:
Question: Since the data from `src_iter` is being accumulated directly into a buffer typed to match `DSTTYPE`, should this be typed as `ImageBuf::ConstIterator<SRCTYPE, DSTTYPE>` instead?

Reply:
I realized that this should probably actually be `ImageBuf::ConstIterator<SRCTYPE, Acc_t>`.

Reply:
yes, exactly. Treat float (or double, for doubles) as the type you do all the real math in, and then the iterators just convert between the buffer format and float.

Reply:
Thanks for confirming.